ipynb/primary-processing/2-filter_sequences_for_blooms.md

```diff
@@ -0,0 +1,102 @@
+```
+
+This next call will setup and verify the path to the bloom sequences used for filtering.
+
+```python
+>>> bloom_sequences = agenv.get_bloom_sequences()
+```
+
+Now let's setup the paths to the sequences to filter. We need the metadata as well in order to reduce the data to just the fecal samples. Please replace these variables with your own paths if you wish to filter your data for blooms (a necessary precursor if you wish to combine data with the American Gut).
+
+```python
+>>> # If you are filtering your own data, please update these filepath variables as necessary
+... sequences = agenv.get_existing_path(agenv.paths['raw-sequences'])
+>>> metadata  = agenv.get_existing_path(agenv.paths['raw-metadata'])
+```
+
+We also need to specify what specific metadata category and value indicate which samples are fecal. It is possible that these values are study specific, so please modify these as needed if you filtering other datasets.
+
+```python
+>>> # If you are filtering your own data, please update these variables to reflect your mapping file
+... metadata_category = 'BODY_SITE'
+>>> metadata_value    = 'UBERON:feces'
+```
+
+Now that we know what sequences to focus on, we can filter the input data down to just those that need to be considered for filtering.
+
+```python
+>>> _fecal_states = ':'.join([metadata_category, metadata_value])
+...
+>>> !filter_fasta.py -f $sequences \
+...                  -o $fecal_sequences \
+...                  --mapping_fp $metadata \
+...                  --valid_states $_fecal_states
+```
+
+The next thing we need to do is setup the parameters for SortMeRNA, which is the method we'll use to compare all the input data to our reference of bloom sequences.
+
+```python
+>>> _params_file = agu.get_path('sortmerna_pick_params.txt')
+>>> with open(_params_file, 'w') as f:
+...     f.write("pick_otus:otu_picking_method sortmerna\n")
+...     f.write("pick_otus:similarity 0.97\n")
+...     f.write("pick_otus:threads %d\n" % multiprocessing.cpu_count())
+...
+>>> !pick_closed_reference_otus.py -i $fecal_sequences \
+...                                -o $observed_blooms \
+...                                -r $bloom_sequences \
+...                                -p $_params_file
+```
+
+And now, we can remove the blooms from the input sequences.
+
+```python
+>>> !filter_fasta.py -f $sequences \
+...                  -m $observed_blooms_otu_map \
+...                  -n \
+...                  -o $filtered_sequences
+```
+
+As the data have now been filtered for blooms, we can now trim the reads back to 100nt to minimize a potential study effect when combining with the Global Gut.
+
+```python
+>>> with open(filtered_sequences) as in_, open(filtered_sequences_100nt, 'w') as out:
+...     agu.trim_fasta(in_, out, 100)
+```
+
+Finally, let's do a quick sanity check that we have sequence data and we'll also dump out summary information about how many reads per sample recruited to the blooms.
```

Comment:
What I don't like about this is the extensive amount of time it'll take to walk over the 2 x 30GBs of sequence data.